src/Header.js

```diff
@@ -0,0 +1,129 @@
+            setDateMessage("Date should be greater than current date")
+            setInputClass("border-red-500");
+            return;
+        }
+
+        setShowImage(true);
+
+        db.collection("todos")
+            .add({
+                todo: inputVal,
+                timestamp: firebase.firestore.FieldValue.serverTimestamp(),
+                date: inputDate,
+                time: inputTime,
+            })
+            .then(() => {
+                // Handle success
+            })
+            .catch((error) => {
+                console.error("Error adding document: ", error);
+            })
+            .finally(() => {
+                setShowImage(false);
+                setInputVal("");
+                setDateMessage("");
+                setErrorMessage("");
+                setTimeMessage("");
+            });
+    };
+
+    const date = new Date();
+    const year = date.getFullYear();
+    const month = String(date.getMonth() + 1).padStart(2, '0');
+    const day = String(date.getDate()).padStart(2, '0');
+    const formattedDate = `${year}-${month}-${day}`;
+
+    const now = new Date();
+    const currentTime = now.toLocaleTimeString();
+
+    return (
+        <>
+            <div className='main-div'>
+                <div className='input-card'>
+                    <div className='header'>
+                        <h3 className='heading'>Add your todo</h3>
+                    </div>
+                    {showImage ?
+                        <div className='loader-div'>
+                            <TbLoader className='loader' />
+                        </div>
+                        :
+                        <div className='input-todo'>
+                            <form>
+                                <input
+                                    className={`mt-1 block w-full input-area px-3 py-2 bg-white border ${inputClass} border-slate-300
+                                     rounded-md text-sm shadow-sm placeholder-slate-400 w-full mt-4
+                                     focus:outline-none focus:border-sky-500 focus:ring-1 focus:ring-sky-500
+                                     disabled:bg-slate-50 disabled:text-slate-500 disabled:border-slate-200 disabled:shadow-none
+                                     invalid:border-pink-500 invalid:text-pink-600
+                                     focus:invalid:border-pink-500 focus:invalid:ring-pink-500`}
+                                    onChange={(e) => setInputVal(e.target.value)}
+                                    placeholder="Enter something..."
+                                    value={inputVal}
+                                />
+
+                                {inputVal.trim() === "" && errorMessage && (
+                                    <div className="error-message ">{errorMessage}</div>
+                                )}
+                                {(formattedDate <= inputDate) ?
```

Comment:
{(formattedDate <= inputDate) ? how is comparison is working? is formattedDate and input date is a number? @Kunal-inno